go/libkb/passphrase_state.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+func randomPassphraseToState(hasRandomPassphrase bool) keybase1.PassphraseState {
+	if hasRandomPassphrase {
+		return keybase1.PassphraseState_RANDOM
+	}
+	return keybase1.PassphraseState_KNOWN
+}
+
+func LoadPassphraseState(mctx MetaContext) (passphraseState keybase1.PassphraseState, err error) {
+	return LoadPassphraseStateWithForceRepoll(mctx, false)
+}
+
+func LoadPassphraseStateWithForceRepoll(mctx MetaContext, forceRepoll bool) (passphraseState keybase1.PassphraseState, err error) {
+	mctx = mctx.WithLogTag("PPSTATE")
+	defer mctx.TraceTimed(fmt.Sprintf("LoadPassphraseState(forceRepoll=%t)", forceRepoll), func() error { return err })()
+
+	if !mctx.G().ActiveDevice.Valid() {
+		mctx.Debug("LoadPassphraseState: user is not logged in")
+		return passphraseState, NewLoginRequiredError("LoadPassphraseState")
+	}
+
+	if !forceRepoll {
```

Comment:
this is unused right now. service bg loop calls LoadFromRemote directly. i can get rid of it